### PR TITLE
Only compute Obligation `cache_key` once  in `register_obligation_at`

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

**Comment by jyn514** [comment]
Perf looks like noise :/

**Comment by petrochenkov** [comment]
Still makes sense as a minor cleanup.
@bors r+ rollup

File: compiler/rustc_data_structures/src/obligation_forest/mod.rs [hunk 1]
```diff
@@ -365,7 +366,7 @@ impl<O: ForestObligation> ObligationForest<O> {
                     && self
                         .error_cache
                         .get(&obligation_tree_id)
-                        .map(|errors| errors.contains(&obligation.as_cache_key()))
+                        .map(|errors| errors.contains(&cache_key))
```
Comment (tamird) [inline compiler/rustc_data_structures/src/obligation_forest/mod.rs:369]: this can be v.key(), allowing the clone on line 345 to be elided.

Comment (tamird) [reply compiler/rustc_data_structures/src/obligation_forest/mod.rs:369]: https://github.com/rust-lang/rust/pull/85436

**Pull request merged**